source/common/singleton/manager_impl.cc

```diff
@@ -0,0 +1,26 @@
+#include "common/singleton/manager_impl.h"
+
+#include "envoy/registry/registry.h"
+
+namespace Envoy {
+namespace Singleton {
+
+InstancePtr ManagerImpl::get(const std::string& name) {
+  verifyRegistration(name);
```

Comment:
My intention was for this to be used by the main thread only, though it doesn't really matter. I can either add appropriate asserts or add locking? Which do you prefer?